java/arcs/core/host/ArcHostContext.kt

```diff
@@ -40,7 +41,8 @@ data class ArcHostContext(
     var arcId: String,
     var particles: MutableMap<String, ParticleContext> = mutableMapOf(),
     var arcState: ArcState = ArcState.NeverStarted,
-    var entityHandleManager: EntityHandleManager
+    var entityHandleManager: EntityHandleManager,
+    var onArcStateChangeCallback: MutableList<(ArcId, ArcState) -> Unit> = mutableListOf()
```

Comment:
The name for this field seems misleading. It's a list of callbacks, not just one.